From ab72766922ca3f1a933592c32eef90cf94c75751 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 19 Jul 2011 16:04:43 +1000
Subject: [PATCH 6/6] Don't assign a static name, strdup it.

pInfo->name is freed in xf86DeleteInput(), make sure it's not a constant
string.

RHEL6-commit only. Upstream has fixed the type/name assignment but to avoid
changed behaviours on RHEL6, the old behaviour is maintained.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/xf86Fpit.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/xf86Fpit.c b/src/xf86Fpit.c
index 5e9e593..a156745 100644
--- a/src/xf86Fpit.c
+++ b/src/xf86Fpit.c
@@ -527,7 +527,7 @@ static int xf86FpitAllocate(InputDriverPtr drv, InputInfoPtr pInfo)
 	priv->fpitOldProximity = 0;
 	priv->fpitIndex = 0;
 	priv->fpitPassive = 0;
-	pInfo->name = XI_TOUCHSCREEN;
+	pInfo->name = strdup(XI_TOUCHSCREEN);
 	pInfo->flags = 0 /* XI86_NO_OPEN_ON_INIT */ ;
 	pInfo->device_control = xf86FpitControl;
 	pInfo->read_input = xf86FpitReadInput;
@@ -579,7 +579,7 @@ static int xf86FpitInit(InputDriverPtr drv, InputInfoPtr pInfo, int flags)
 		pInfo->fd = -1;
 	}
 	priv->fpitDev = strdup(str);
-	pInfo->name = xf86SetStrOption(pInfo->options, "DeviceName", XI_TOUCHSCREEN);
+	pInfo->name = xf86SetStrOption(pInfo->options, "DeviceName", pInfo->name);
 	xf86Msg(X_CONFIG, "FPIT device name: %s\n", pInfo->name);
 	priv->screen_no = xf86SetIntOption(pInfo->options, "ScreenNo", 0);
 	xf86Msg(X_CONFIG, "Fpit associated screen: %d\n", priv->screen_no);
-- 
1.7.6

